code/modules/assembly/igniter.dm

```diff
@@ -26,16 +26,17 @@
 	var/turf/location = get_turf(loc)
 	if(location)
 		location.hotspot_expose(1000,1000)
+	sparks.start()
 	if(istype(loc, /obj/item/assembly_holder))
-		if(istype(loc.loc, /obj/structure/reagent_dispensers/fueltank))
-			var/obj/structure/reagent_dispensers/fueltank/tank = loc.loc
+		var/locloc = loc.loc
+		if(istype(locloc, /obj/structure/reagent_dispensers/fueltank))
+			var/obj/structure/reagent_dispensers/fueltank/tank = locloc
 			if(tank)
-				tank.boom(TRUE)
-		if(istype(loc.loc, /obj/item/reagent_containers/glass/beaker))
-			var/obj/item/reagent_containers/glass/beaker/beakerbomb = loc.loc
+				tank.boom(TRUE)  // this qdel's `src`
+		else if(istype(locloc, /obj/item/reagent_containers/glass/beaker))
```

Comment:
did you mean to make this `locloc` ?

Reply:
yes, i did.

The original code goes like
```js
if(istype(loc.loc, ..))
  go_boom_and_qdel_ourselves()
if(istype(loc.loc, ..)) // runtime here, because loc is null and loc.loc is inaccessible
```
and there are two ways to address: 1) cache the `loc.loc` so that it still holds a meaningful value by the time of second `if`, and 2) change the second `if` to `else if`.
I couldn't decide on which one is better here, so I applied both of those at once.